### ISSUE TITLE
Leveling Suggestion

### ISSUE THREAD

**Issue by TheExceed** [drtshock/Factions#521]
I am sure you have played some type of game similar to Clash of Clans, where you have to level up your builds and whatnot, is it possibly to add this type of leveling to factions? Gaining experience to level your faction can be anything such as: killing enemy faction members and when you want to level up it would take a configured amount of money from the faction bank and leveling up will give enhanced features: increase member cap, increase bank cap, etc.

I really think this would be a sweet addition.

**Comment by drtshock** [comment]
So ELO for factions? Other servers have done this like Daegonner. We would just have to figure out a way to handle that for raids. Would require adding a core or something along the lines of that.

**Comment by Agazed** [comment]
Cool idea, but this would take tons of work.

**Comment by Kakifrucht** [comment]
And bloat the plugin too much imho.

**Comment by TheExceed** [comment]
I mean it would allow for more player interaction. Right now, you make a faction, and raid... not hard to build a faction up with alts and get a nice large base. This would make it a bit more difficult to make a huge faction quickly. Where they have to slowly build themselves up. I hope that it actually can be considered. :)

**Comment by TheExceed** [comment]
I mean I wouldnt mind pay a bit extra for this feature to be separate. :)

**Comment by DahFrozen** [comment]
I believe this feature could have the potential to bring factions "alive" again. Seeing something like this implemented into factions will explode in the community especially as the new 1.9 "Combat" update is coming early next year.

You can add a levelling system were your faction gains EXP for doing certain tasks. Upon reaching configurable levels, your faction unlocks new enhancements that other factions don't have. The faction's level progress can be lost when it is disbanded - this keeps players playing to prevent the faction being removed due to inactivity. It would also add bragging rights.

Enhancement suggestions:
- Bigger faction vaults (Like a chest access with a command that only faction members can access)
- Bigger power/claim/member limit.
- The potion effects from beacons only effect your faction members when they are inside the faction's claimed land.
- EXP multipliers (Faction members will gain extra EXP from mob grinding and stuff at a certain faction level) 
- Mob drops bonus (Factions with a high level can find bonus mob drops while grinding)

**Comment by TheExceed** [comment]
My suggestion for possible level perks: Used from @DahFrozen 
When suggesting these, I also suggest that they can be easily disabled too. :)

-Custom Faction chest (each faction rank would increase the size)
-Increased power/claim/members (each faction rank would increase the size)
-Exp Bonus Faction members will gain extra EXP from mob grindin at a certain faction level)
-Temporary/permanent potion effects (Each faction rank can add/upgrade the effects) (only in faction claims not wilderness)
-Faction warps (each faction rank increases the amount and decreases the price) 
-Rewards (Upon ranking up, leader/faction gets rewards, anything with a command)
-Faction mobs (currently an add on) (Each faction increase number and strength/health)

**Comment by drtshock** [comment]
Yeah a lot of new features to go with 1.9 would be nice. _looks around to see if anyone wants to help implement them_

**Comment by DahFrozen** [comment]
I would help if I knew how to code :S sorry xD
